arcgis-ios-sdk-samples/Content Display Logic/Controllers/ContentTableViewController.swift

```diff
@@ -14,40 +14,74 @@
+class ContentTableViewController: UITableViewController {
+    
     var samples = [Sample]()
     private var expandedRowIndex:Int = -1
     
-    private var headerView:CustomSearchHeaderView!
     var containsSearchResults = false
     private var bundleResourceRequest:NSBundleResourceRequest!
     private var downloadProgressView:DownloadProgressView!
-
-    var token: Int = 0
+    
+    var searchEngine: SearchEngine?
+    
+    // strong reference needed for iOS 10
+    var filterSearchController:UISearchController?
     
     override func viewDidLoad() {
         super.viewDidLoad()
         
-        self.tableView.rowHeight = UITableView.automaticDimension
-        self.tableView.estimatedRowHeight = 60
-        
-        if containsSearchResults {
-            self.tableView.tableHeaderView?.removeFromSuperview()
-            self.tableView.tableHeaderView = nil
-        }
-        else {
-            self.headerView = self.tableView.tableHeaderView! as? CustomSearchHeaderView
-            self.headerView.delegate = self
-            self.headerView.hideSuggestionsTable()
-        }
+        searchEngine = SearchEngine(samples: samples)
         
         //initialize download progress view
-        self.downloadProgressView = DownloadProgressView()
-        self.downloadProgressView.delegate = self
+        downloadProgressView = DownloadProgressView()
+        downloadProgressView.delegate = self
+        
+        if !containsSearchResults{
+            addFilterSearchController()
+        }
     }
     
-    func samplesByNames<C: Collection>(_ names: C) -> [Sample] where C.Element == String {
-        return samples.filter { names.contains($0.name) }
+    private func addFilterSearchController(){
+        
+        // ensure that the search results are interactable
+        definesPresentationContext = true
+        
+        // create the search controller
+        let searchController = UISearchController(searchResultsController:nil)
+        searchController.obscuresBackgroundDuringPresentation = false
+        searchController.hidesNavigationBarDuringPresentation = false
+        // send search query updates to the results controller
+        searchController.searchResultsUpdater = self
+        // retain a strong reference for iOS 10
+        self.filterSearchController = searchController
+        
+        let searchBar = searchController.searchBar
+        searchBar.placeholder = "Filter"
+        searchBar.autocapitalizationType = .none
+        // set the color of "Cancel" text
+        searchBar.tintColor = .white
+        
+        if #available(iOS 11.0, *) {
+            // embed the search bar under the title in the navigation bar
+            navigationItem.searchController = searchController
+            
+            // find the text field to customize its appearance
+            if let textfield = searchBar.value(forKey: "searchField") as? UITextField {
```

Comment:
This is a HUGE hack. Do we really need to modify the search bar's text field? I'd rather we didn't.

Reply:
I spent awhile online trying to find the best way to change the background color of a text field embedded in the navigation bar. It's far more of a hack than we would hope, but this was the least bad solution I could find. Otherwise it looks like this on iOS 11+

![screen shot 2018-09-13 at 3 34 32 pm](https://user-images.githubusercontent.com/42748837/45519599-a3f5a000-b76a-11e8-97d5-3508c63a9f9a.png)

Reply:
That is weird. I wonder where that blue color is coming from?

Reply:
It appears to be a computed color derived from the blue navigation bar. See [this example](https://stackoverflow.com/questions/46007260/ios-11-customise-search-bar-in-navigation-bar) with a very similar circumstance and a solution that uses the hack.

Reply:
It might be better to just remove the navigation bar's custom `barTintColor`. Particularly if we git rid of the blue title bar on macOS. But I think we should get feedback from the rest of the team before getting rid of the blue bar on either platform.

Reply:
I actually don't mind the bar color. In any case, we shouldn't restrict ourselves on the bar color based on the search field.

Reply:
I don't mind the bar color either. What I do mind is hacking Apple provided components. I would much prefer to remove the custom color over relying on implementation details of `UISearchBar`.